src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -537,17 +548,41 @@ def _try_parse_msi_account_name(account):
                 return parts[0], (None if len(parts) <= 1 else parts[1])
         return None, None
 
+    def _create_identity_credential(self, account, aux_tenant_id=None):
+        user_type = account[_USER_ENTITY][_USER_TYPE]
+        username_or_sp_id = account[_USER_ENTITY][_USER_NAME]
+        home_account_id = account[_USER_ENTITY][_HOME_ACCOUNT_ID]
+        identity_type, identity_id = Profile._try_parse_msi_account_name(account)
+        tenant_id = aux_tenant_id if aux_tenant_id else account[_TENANT_ID]
+        from azure.cli.core._credential import IdentityCredential
+        # initialize IdentityCredential
+        if identity_type is None:
+            if in_cloud_console() and account[_USER_ENTITY].get(_CLOUD_SHELL_ID):
+                if aux_tenant_id:
+                    raise CLIError("Tenant shouldn't be specified for Cloud Shell account")
+                return IdentityCredential()
+            if user_type == _USER:
+                authority = self.cli_ctx.cloud.endpoints.active_directory.lstrip('https://')
```

Comment:
It is not safe to use [`lstrip`](https://docs.python.org/3.8/library/stdtypes.html#str.lstrip):

> `str.lstrip([chars])`
> The `chars` argument is a string specifying the set of characters to be removed.

So this can cause unwanted stripping of `hsp`:

```
>>> "https://hsp.myorg.com".lstrip('https://')
'.myorg.com'
```

[This stackoverflow answer](https://stackoverflow.com/a/1038845/2199657) is the correct way to achieve this.

Reply:
Good catch, thanks